Disable idempotent module and fix Rails 6.1 compat issue

jordan-brough opened the pull request:
The idempotent module seems to have a bug around `add_index`.
I was able to replicate by running `rails db:migrate:down` on a migration with a simple `add_index`.

I'm also a little dubious about whether we should enable the idempotency. See https://github.com/doctolib/safe-pg-migrations/issues/60.

Update: Also: `StatementInsurer` had a Rails 6.1 bug.

See individual commits.

jordan-brough committed:
Update configs for our own testing

- Ruby 3.0.2
- Postgres 11 (via Docker)

jordan-brough committed:
Skip tests that are failing on master

jordan-brough committed:
Disable idempotent module

nhippenmeyer reviewed (approved):
Thanks for doing this. I hadn't noticed the bug around adding indexes. What's your thinking wrt commenting out the problematic code vs deleting it?

jordan-brough commented:
Hm, good question. My thoughts, assuming that we'll need to rebase in the future:
- For `lib/safe-pg-migrations/base.rb`: Nice to keep, it's a good spot to have a comment added
- For `lib/safe-pg-migrations/plugins/idem_potent_statements.rb` it's nice to just avoid touching it. E.g. in `master` branch that module got renamed.
- For `test/xxx`: ~It'd probably be better to just delete those. I'll update to do that~ Edit: Actually just `skip`ing those might make future rebasing easiest? I'll try updating to that.

jordan-brough committed:
Skip tests instead of commenting them out

To make future rebases easier

jordan-brough committed:
Remove unnecessary test change

jordan-brough committed:
Rails 6.1 fix for "remove_index"

This was causing an error like:

    ArgumentError: You must specify the index name

When running "rails db:migrate:down" on a migration like:

    add_index :users, :email, unique: true

jordan-brough commented:
@nhippenmeyer I updated to `skip` instead of comment-out, and I also added another Rails 6.1 fix. Look good still?
